builtInFunctions/multiESDTNFTTransfer_test.go

```diff
@@ -1193,3 +1193,221 @@ func runMultiTransfer(t *testing.T, isScToScEventLogEnabled bool) (*vmcommon.VMO
+
+	return vmInput, multiTransfer
+}
+
+func TestESDTNFTMultiTransfer_ProcessBuiltinFunctionOnSameShardWithScCallWithEGLDNotEnabled(t *testing.T) {
+	t.Parallel()
+
+	vmInput, multiTransfer := createSetupForMultiTransferWithEGLD(t)
+	multiTransfer.enableEpochsHandler = &mock.EnableEpochsHandlerStub{
+		IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+			return flag != EGLDInESDTMultiTransferFlag
+		},
+	}
+
+	sender, err := multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err := multiTransfer.accounts.LoadAccount(vmInput.Arguments[0])
+	require.Nil(t, err)
+
+	_, err = multiTransfer.ProcessBuiltinFunction(sender.(vmcommon.UserAccountHandler), destination.(vmcommon.UserAccountHandler), vmInput)
+	require.NotNil(t, err)
+}
+
+func TestESDTNFTMultiTransfer_ProcessBuiltinFunctionOnSameShardWithScCallWithEGLD(t *testing.T) {
+	t.Parallel()
+
+	vmInput, multiTransfer := createSetupForMultiTransferWithEGLD(t)
+	multiTransfer.enableEpochsHandler = &mock.EnableEpochsHandlerStub{
+		IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+			return true
+		},
+	}
+
+	sender, err := multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err := multiTransfer.accounts.LoadAccount(vmInput.RecipientAddr)
+	require.Nil(t, err)
+
+	vmOutput, err := multiTransfer.ProcessBuiltinFunction(sender.(vmcommon.UserAccountHandler), destination.(vmcommon.UserAccountHandler), vmInput)
+	require.Nil(t, err)
+	require.Equal(t, vmcommon.Ok, vmOutput.ReturnCode)
+
+	_ = multiTransfer.accounts.SaveAccount(sender)
+	_, _ = multiTransfer.accounts.Commit()
+
+	// reload accounts
+	sender, err = multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err = multiTransfer.accounts.LoadAccount(vmInput.Arguments[0])
+	require.Nil(t, err)
+
+	testNFTTokenShouldExist(t, multiTransfer.marshaller, sender, []byte("token1"), 1, big.NewInt(2)) // 3 initial - 1 transferred
+	testNFTTokenShouldExist(t, multiTransfer.marshaller, destination, []byte("token1"), 1, big.NewInt(4))
+
+	require.Equal(t, sender.(vmcommon.UserAccountHandler).GetBalance(), big.NewInt(2))
+	require.Equal(t, destination.(vmcommon.UserAccountHandler).GetBalance(), big.NewInt(1))
+
+	funcName, args := extractScResultsFromVmOutput(t, vmOutput)
+
+	scCallFunctionAsHex := hex.EncodeToString([]byte("functionToCall"))
+	scCallArg := hex.EncodeToString([]byte("arg"))
+	assert.Equal(t, scCallFunctionAsHex, funcName)
+	require.Equal(t, 1, len(args))
+	require.Equal(t, []byte(scCallArg), args[0])
+}
+
+func TestESDTNFTMultiTransfer_ProcessBuiltinFunctionOnCrossShardsWithEGLD(t *testing.T) {
+	t.Parallel()
+
+	payableHandler := &mock.PayableHandlerStub{
+		IsPayableCalled: func(address []byte) (bool, error) {
+			return true, nil
+		},
+	}
+
+	multiTransferSenderShard := createESDTNFTMultiTransferWithMockArguments(1, 2, &mock.GlobalSettingsHandlerStub{})
```

Comment:
Is there any way to put all this setup code in a func similar to `createSetupForMultiTransferWithEGLD` for cross shard?

Reply:
yes, it would mean to recheck multiple tests.